plugins/outputs/orionres/res/schema/schema.go

```diff
@@ -0,0 +1,183 @@
+	// specified (all name filters specified must match for the request to succeed)
+	if r.schemaType == info.SetTypeRes {
+		resSets, nonMatchedFilters := r.listResSet(nameFilters)
+		if len(nonMatchedFilters) > 0 {
+			err := fmt.Errorf("The following name filter(s) did not match anything: %s",
+				strings.Join(nonMatchedFilters, ", "))
+			return http.StatusBadRequest, luddite.NewError(errorDefs, EcodeLookupFailure, err)
+		}
+		return http.StatusOK, resSets
+	}
+
+	// List the DimSet objects - limited to the specified names if filters have been
+	// specified (all name filters specified must match for the request to succeed)
+	dimSets, nonMatchedFilters := r.listDimSet(nameFilters)
+	if len(nonMatchedFilters) > 0 {
+		err := fmt.Errorf("The following name filter(s) did not match anything: %s",
+			strings.Join(nonMatchedFilters, ", "))
+		return http.StatusBadRequest, luddite.NewError(errorDefs, EcodeLookupFailure, err)
+	}
+	return http.StatusOK, dimSets
+}
+
+func (r *schemaResource) Count(req *http.Request) (int, interface{}) {
+	return http.StatusOK, 0
+}
+
+func (r *schemaResource) Get(req *http.Request, name string) (int, interface{}) {
+	// Attempt to get the requested ResSet definition
+	if r.schemaType == info.SetTypeRes {
+		resSets := r.processor.MetricDefs.Res
+		if resSet, ok := resSets[name]; ok {
+			return http.StatusOK, resSet
+		}
+		return http.StatusNotFound, nil
+	}
+
+	// Attempt to get the requested DimSet definition
+	dimSets := r.processor.MetricDefs.Dim
+	if dimSet, ok := dimSets[name]; ok {
+		return http.StatusOK, dimSet
+	}
+	return http.StatusNotFound, nil
+}
+
+func (r *schemaResource) Create(req *http.Request, value interface{}) (int, interface{}) {
+	return http.StatusNotImplemented, nil
+}
+
+func (r *schemaResource) Delete(req *http.Request, id string) (int, interface{}) {
+	return http.StatusNotImplemented, nil
+}
+
+func (r *schemaResource) Update(req *http.Request, id string, value interface{}) (int, interface{}) {
+	return http.StatusNotImplemented, nil
+}
+
+func (r *schemaResource) Action(req *http.Request, id string, action string) (int, interface{}) {
+	return http.StatusNotFound, nil
+}
+
+func (r *schemaResource) listResSet(filters []string) ([]*info.ResSet, []string) {
+	resSets := r.processor.MetricDefs.Res
+
+	var ret []*info.ResSet
+	var resSet *info.ResSet
+	var nonMatchedFilters []string
+
+	// If filters are supplied, only report ResSet's matched
+	filterCount := len(filters)
+	if filterCount > 0 {
+		var ok bool
+		var filter string
+		ret = make([]*info.ResSet, 0, filterCount)
+		for _, filter = range filters {
+			if resSet, ok = resSets[filter]; ok {
+				ret = append(ret, resSet)
+			} else {
+				nonMatchedFilters = append(nonMatchedFilters, filter)
+			}
+		}
+		return ret, nonMatchedFilters
+	}
+
+	// Report all ResSets available
+	ret = make([]*info.ResSet, 0, len(resSets))
+	for _, resSet = range resSets {
+		ret = append(ret, resSet)
+	}
+	return ret, nonMatchedFilters
```

Comment:
I think the code looks good.  Though I'm crossing my fingers a little that ResSet has everything needed for the python code.  I'm sure it has enough schema for orion-res API but hopefully not missing something used by LCM.

Reply:
Hmmm. I thought they would be the same definitions. If they need more, do you see any reason why we could not extend the definitions in the go code? Would that break anything? I would expect that it would just add additional fields that would be ignored by anything currently using those definitions (unless there is a json schema being applied somewhere).

Reply:
I think it might... If it doesn't you may have to add.  It shouldn't break anything.

Reply:
I will hold off on this merge until I can run a test with a hacked test-case and see it work. If it does not work, obviously I have more work to do (though I would not think it would be too much).